api/src/main/java/jakarta/persistence/criteria/CriteriaBuilder.java

```diff
@@ -702,7 +732,52 @@
      */	
     Expression<Double> sqrt(Expression<? extends Number> x);
 
-	
+    /**
+     * Create an expression that returns the exponential
+     * of its argument, that is, Euler's number <i>e</i>
+     * raised to the power of its argument.
+     * @param x expression
+     * @return exponential
+     */
+    Expression<Double> exp(Expression<? extends Number> x);
+
+    /**
+     * Create an expression that returns the natural logarithm
+     * of its argument.
+     * @param x expression
+     * @return natural logarithm
+     */
+    Expression<Double> ln(Expression<? extends Number> x);
+
+    /**
+     * Create an expression that returns the first argument
+     * raised to the power of its second argument.
+     * @param x base
+     * @param y exponent
+     * @return the base raised to the power of the exponent
+     */
+    Expression<Double> power(Expression<? extends Number> x, Expression<? extends Number> y);
+
+    /**
+     * Create an expression that returns the first argument
+     * raised to the power of its second argument.
+     * @param x base
+     * @param y exponent
+     * @return the base raised to the power of the exponent
+     */
+    Expression<Double> power(Expression<? extends Number> x, Number y);
+
+    /**
+     * Create an expression that returns the first argument
+     * rounded to the number of decimal places given by the
+     * second argument.
+     * @param x base
+     * @param n number of decimal places
+     * @return the rounded value
+     */
+    <T extends Number> Expression<T> round(Expression<T> x, Integer n);
```

Comment:
This can be added too:
```
<T extends Number> Expression<T> round(Expression<T> x, Expression<T> n);
```

Reply:
I mean, it can certainly add it, but I'm kinda struggling I bit to see what someone would use it for...

Reply:
At least 1st one makes sense. I have some issues with testing this on PostgreSQL - https://www.postgresql.org/docs/9.5/functions-math.html
There are 2 prototypes:
- round(dp or numeric) - supports dp
- round(v numeric, s int) - numeric values only, throws `Internal Exception: org.postgresql.util.PSQLException: ERROR: function round(double precision, integer) does not exist` for Java Double in entity.

2nd prototype makes sense when both values will come from Entity/table, like in my jUnit for POWER:
```
public class NumberEntity {
    @Id
    private Integer id;
    private Long longValue;
    private Double doubleValue;
   ...
}

        CriteriaBuilder cb = em.getCriteriaBuilder();
        CriteriaQuery<Double> cq = cb.createQuery(Double.class);
        Root<NumberEntity> number = cq.from(NumberEntity.class);
        cq.select(cb.power(number.get("doubleValue"), number.get("longValue")));
        cq.where(cb.equal(number.get("id"), id));
        return em.createQuery(cq).getSingleResult();
```

Reply:
Actually perhaps `cast(cast(d as decimal(20,2)) as double precision)` is better in some contexts.

Reply:
Or you can use `floor()`.

Reply:
Alright, so I just tested this across all platforms we support, and this is indeed an issue very specific to Postgres. I think the most reasonable solution is to emulate `round(x,n)` using `floor(x*10^n+0.5)/10^n` on Postgres, at least when the argument is a double precision type.